### DISCUSSION TITLE
Why does my dynamic image not display in child component using SSR in Nuxt and Quasar?

### DISCUSSION THREAD

**Discussion by s-spoon** [quasarframework/quasar#15870] [CLI - SSR mode] [answered]
Urgent help is needed for my SSR projects (Nuxt, Quasar). When I try to display a dynamic image in a child component, the image is not displayed.

My current code is:

```
function getUrl (img: string)  {
  return require(`@/assets/imgs/sidebar/${img}.png`);
}
```

It shows error message

_Cannot find module '@/assets/imgs/sidebar/side_1.png.png' Require stack: - E:\Work\0503_game777\nuxt-quasar\components\sidebar\SideBarItem.vue_

I don't know what is the reason but the image path is correct and also it shows when I use the path directly.

And also I tried like this:

```
function getUrl (img: string)  {
  return `/_nuxt/assets/imgs/sidebar/${img}`;
}
```
It works locally but, when I built it, then it doesn't show the image.

Please somebody help me. Thanks

I tried to get dynamic image path with "import" instead of "require()" but is shows the promise error. My current code is typescript.

**Comment by dongwa** [comment, marked as answer]
In Quasar, it is better to reference local images in the public folder. Please refer to https://quasar.dev/quasar-cli-vite/handling-assets for more information.

**Reply by s-spoon** [reply]
Thanks dongwa, that works in my case. I didn't think about that. :)
